src/lib.rs

```diff
@@ -1,110 +1,451 @@
-
-        self.content.push(Section { title, content });
-        self
+    /// Append a text line.
+    ///
+    /// The line will be rendered in a way that ensures it can't be
+    /// interpreted as a control line. The caller does not need to
+    /// ensure, for example, that the line doesn't start with a
+    /// period ("`.`") or an apostrophe ("`'`").
+    pub fn text(&mut self, inlines: impl Into<Vec<Inline>>) {
+        self.lines.push(Line::text(inlines.into()));
     }
-}
 
-impl Troffable for Roff {
-    fn render(&self) -> String {
-        let mut res = String::new();
+    /// Render as ROFF source text that can be fed to a ROFF implementation.
+    pub fn render(&self) -> String {
+        let mut buf = vec![];
+        self.to_writer(&mut buf).unwrap(); // writing to a Vec always works
+        String::from_utf8_lossy(&buf).into_owned()
+    }
 
-        writeln!(
-            &mut res,
-            ".TH {} {}",
-            self.title.to_uppercase(),
-            self.section
-        )
-        .unwrap();
-        for section in &self.content {
-            writeln!(&mut res, "{}", escape(&section.render())).unwrap();
+    /// Write to a writer.
+    pub fn to_writer(&self, w: &mut dyn Write) -> Result<(), std::io::Error> {
+        w.write_all(APOSTROPHE_PREABMLE.as_bytes())?;
+        for line in self.lines.iter() {
+            line.render(w, Apostrophes::Handle)?;
         }
+        Ok(())
+    }
 
-        res
+    /// Render without handling apostrophes specially.
+    ///
+    /// You probably want [`render`](Roff::render) or
+    /// [`to_writer`](Roff::to_writer) instead of this method.
+    ///
+    /// Without special handling, apostrophes get typeset a right
+    /// single quotes, including in words like "don't". In most
+    /// situations, such as in manual pages, that's unwanted. The
+    /// other methods handle apostrophes specially to prevent it, but
+    /// for completeness, and for testing, this method is provided to
+    /// avoid it.
+    pub fn to_roff(&self) -> String {
+        let mut buf = vec![];
+        for line in self.lines.iter() {
+            // Writing to a Vec always works, so we discard any error.
+            line.render(&mut buf, Apostrophes::DontHandle).unwrap();
+        }
+        String::from_utf8_lossy(&buf).into_owned()
     }
 }
 
-#[derive(PartialEq, Eq)]
-struct Section {
-    title: String,
-    content: String,
+/// Build a Roff.
+#[derive(Default)]
+pub struct Builder {
+    roff: Roff,
 }
 
-impl Troffable for Section {
-    fn render(&self) -> String {
-        let mut res = String::new();
+impl Builder {
+    /// Append a control line.
+    ///
+    /// The line consist of the name of a built-in command or macro,
+    /// and some number of arguments. Arguments that contain spaces
+    /// will be enclosed on double quotation marks.
+    pub fn control<'a>(mut self, name: &'a str, args: impl Into<Vec<&'a str>>) -> Self {
```

Comment:
Do we need a separate builder or should we just make the `Roff` functions return `-> &mut Self`?

Reply:
I found the separate builder more convenient in cases where the whole page can't be built at once.

Reply:
I feel like I'm missing something.  Effectively the only difference is one would be a `&mut self` builder and the other is a `self` builder.